### PR TITLE
feat: Minimal in-browser compatibility for the crate

### PR DESCRIPTION
**Pull request by CPerezz**
This changes the associated type `Params` of the `XXXFCircuit` to use
the newly created `PathOrBin` type.

This allows the user of the lib to directly send the binary of the files
already read or instead, provide a path to it and let `sonobe` do the
work.

With this, Circom and Noir frontends should be already usable from the browser if we allow
JS to take care of reading the `.wasm` and `.r1cs` files.

This currently depends on:
- https://github.com/dmpierre/arkworks_backend/pull/1
- https://github.com/arnaucube/circom-compat/pull/3

### CODE REVIEW

**Comment by CPerezz** [comment]
Submitted [3e2f8fc](https://github.com/dmpierre/arkworks_backend/pull/1/commits/3e2f8fc00a0ca663f130f49c315b6a71c6813b66)

Will work on the top of the rev for the time being. But not merge until we have it all working and merged.

**Comment by CPerezz** [comment]
This is awaiting feedback from @czhang2718 and @janabel to see if with this is feasible and decently OK to use the crate compiled to WASM from the browser.

**Comment by CPerezz** [comment]
Got confirmation 0XPARC team that WASM is working. So this is good to go!
Will update the dependencies to `branch` instead of `rev`.

**Review by dmpierre — APPROVED** [review]
Nice work, excited for finally having wasm compilation!

**Pull request merged**
